tools: add PIDFile option in frr.service

emabo opened the pull request:
when type is forking, it is recommended to also use the PIDFile= option,
so that systemd can reliably identify the main process of the service.

ton31337 commented on tools/frr.service:
```diff
@@ -17,6 +17,7 @@ WatchdogSec=60s
 RestartSec=5
 Restart=on-abnormal
 LimitNOFILE=1024
+PIDFile=/var/run/frr/watchfrr.pid
```
What happens when watchfrr is disabled?

ton31337 reviewed (changes requested):
I'm not really sure about this change. What is we have a different rundir?

emabo commented:
> I'm not really sure about this change. What is we have a different rundir?

I partially agree, I used the default path, like ExecStart, ExecStop and ExecReload.
Perphaps, we should include this file in the compilation process like frr.in, replacing the current and correct paths.

ton31337 commented:
Even more, `RuntimeDirectory` is recommended if you create pid files in sub-directory, to avoid permission problems.

donaldsharp commented:
Any change here must respect the run directory as configured on the run line.  This will break people who have non-standard run directories

donaldsharp commented:
What would happen if I ran a configure with a `--localstatedir=/tmp/frr`. This change would break that.  If someone does this the frr.service file installed should do the right thing.

emabo commented:
> What would happen if I ran a configure with a `--localstatedir=/tmp/frr`. This change would break that. If someone does this the frr.service file installed should do the right thing.

I changed my commit in order to keep into consideration  --localstatedir and --sbindir.
The two files frr.service and frr@.service are now regenerated for each configure command, but they aren't installed with make install, as before.
I adjusted also frr@.service in order to consider %I parameter.

emabo commented:
> Even more, `RuntimeDirectory` is recommended if you create pid files in sub-directory, to avoid permission problems.

IMHO I think it's not needed, because RuntimeDirectory is already created with correct permissions in daemon_start() function (frrcommon.sh.in) when watchfrr is launched by systemd.

ton31337 reviewed (approved):
LGTM
